### PR TITLE
feat(aad/rules): add new forward rule resource

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Support the new forward rule resource for the Advanced Anti-DDoS.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. support forward rule resource for the Advanced Anti-DDoS.
2. support the related ducument and acc test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/aad' TESTARGS='-run=TestAccForwardRule_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/aad -v -run=TestAccForwardRule_basic -timeout 360m -parallel 4
=== RUN   TestAccForwardRule_basic
=== PAUSE TestAccForwardRule_basic
=== CONT  TestAccForwardRule_basic
--- PASS: TestAccForwardRule_basic (19.45s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/aad       19.582s
```

### CODE REVIEW

File: huaweicloud/services/aad/resource_huaweicloud_aad_forward_rule.go [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+		ReadContext:   resourceForwardRuleRead,
+		UpdateContext: resourceForwardRuleUpdate,
+		DeleteContext: resourceForwardRuleDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceForwardRuleImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/aad/resource_huaweicloud_aad_forward_rule.go]: this parameter is missing in the doc
should it be ForceNew?

File: docs/resources/aad_forward_rule.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  source_port         = 888
+  source_ip_addresses = "1.1.1.1,2.2.2.2"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `instance_id` - (Optional, String, ForceNew) Specifies the ID of advanced Anti-DDoS instance.
```
Comment (Jason-Zhang9309) [inline docs/resources/aad_forward_rule.md]: it's required in the code

File: docs/resources/aad_forward_rule.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  Changing this will create a new rule resource.
+
+* `advanced_ip` - (Required, String, ForceNew) Specifies the public IP addresss to which Advanced Anti-DDoS instance
```
Comment (Jason-Zhang9309) [inline docs/resources/aad_forward_rule.md]: it's `ip` in the code

File: huaweicloud/services/aad/resource_huaweicloud_aad_forward_rule.go [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+				ForwardProtocol: protocol,
+				ForwardPort:     forwardPort,
+				SourcePort:      d.Get("source_port").(int),
+				SourceIp:        d.Get("source_ip").(string),
+			},
+		},
+	}
+	_, err = rules.BatchCreate(client, instanceId, advancedIp, opts)
+	if err != nil {
+		return diag.FromErr(err)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/aad/resource_huaweicloud_aad_forward_rule.go]: add an error message here

File: huaweicloud/services/aad/resource_huaweicloud_aad_forward_rule.go [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+	}
+	tflog.Debug(ctx, "Retrieved Advanced Anti-DDos forward rule: %#v", resp)
+
+	opts := rules.UpdateOpts{
+		ForwardProtocol: d.Get("forward_protocol").(string),
+		ForwardPort:     d.Get("forward_port").(int),
+		SourcePort:      d.Get("source_port").(int),
+		SourceIp:        d.Get("source_ip").(string),
+	}
+	err = rules.Update(client, instanceId, advancedIp, resp.ID, opts)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/aad/resource_huaweicloud_aad_forward_rule.go]: how about save  `resp.ID` in the state file, so you don't need to get it here

File: huaweicloud/services/aad/resource_huaweicloud_aad_forward_rule.go (hunk 4 above)
Comment (Lance52259) [reply huaweicloud/services/aad/resource_huaweicloud_aad_forward_rule.go]: done.

File: huaweicloud/services/aad/resource_huaweicloud_aad_forward_rule.go (hunk 5 above)
Comment (Lance52259) [reply huaweicloud/services/aad/resource_huaweicloud_aad_forward_rule.go]: I don't want to expose a useless param.

File: docs/resources/aad_forward_rule.md (hunk 3 above)
Comment (Lance52259) [reply docs/resources/aad_forward_rule.md]: done.

File: docs/resources/aad_forward_rule.md (hunk 2 above)
Comment (Lance52259) [reply docs/resources/aad_forward_rule.md]: done.

File: huaweicloud/services/aad/resource_huaweicloud_aad_forward_rule.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/aad/resource_huaweicloud_aad_forward_rule.go]: This parameter has been removed.

**Pull request merged**